controllers/proxy/ecr_proxy.go

```diff
@@ -15,9 +16,18 @@ func NewECRProxy(registryEndpoint string) ProxyFunc {
 	})
 
 	return ProxyFunc(func(token string, w http.ResponseWriter, r *http.Request) {
+		originalHost := r.Host
 		r.Header.Set("Authorization", fmt.Sprintf("Basic %s", token))
 		r.Host = registryEndpoint
 
+		reverseProxy.ModifyResponse = func(resp *http.Response) error {
```

Comment:
That's not quite correct, and I want to be clear in case this ever comes up again:

The 301 redirect is a red herring. All it does is redirect (not rewrite) any `http` requests to `https`. Note that this redirect is redundant in our case since the load balancers we use only listen on `https:443` - no `http` requests would even make it to nginx anyway. 

The real behavior nginx had that we didn't was [here](https://gitlab.ims.io/carbon/docker-registry-private-auth0/blob/master/nginx/nginx/nginx.default#L67):
```
  ...
  proxy_redirect {{backend}}/ https://$host/;
}
```

**That** function, `proxy_redirect`, is what writes the `Location` header. Don't confuse the `proxy_redirect` with the http->https redirect!